editor/src/clj/editor/gui.clj

```diff
@@ -3209,24 +3209,28 @@
     :else
     (dissoc node-desc :overridden-fields)))
 
-(defn- make-rt-layout-desc [layout-name decorated-node-msgs]
+(defn- make-rt-layout-desc [layout-name decorated-node-msgs id->node-desc-for-default-layout]
+  {:pre [(map? id->node-desc-for-default-layout)]}
   (protobuf/make-map-without-defaults Gui$SceneDesc$LayoutDesc
     :name layout-name
     :nodes (coll/transfer decorated-node-msgs []
              (keep
                (fn [{:keys [layout->prop->value] :as decorated-node-msg}]
                  {:pre [(map? layout->prop->value)]}
-                 (let [default-prop->value (layout->prop->value "")
-                       prop->value (layout->prop->value layout-name)]
-                   (assert (map? default-prop->value))
+                 (let [prop->value (layout->prop->value layout-name)]
                    (assert (map? prop->value))
-                   (when (not= default-prop->value prop->value)
-                     (some->> (-> decorated-node-msg
-                                  (select-keys [:custom-type :id :parent :template-node-child :type])
-                                  (into (map prop-entry->pb-field-entry)
-                                        prop->value)
-                                  (node-desc->rt-node-desc layout-name))
-                              (protobuf/clear-defaults Gui$NodeDesc)))))))))
+                   (when-some [node-desc-for-layout
+                               (some->> (-> decorated-node-msg
+                                            (select-keys [:custom-type :id :parent :template-node-child :type])
+                                            (into (map prop-entry->pb-field-entry)
+                                                  prop->value)
+                                            (node-desc->rt-node-desc layout-name))
+                                        (protobuf/clear-defaults Gui$NodeDesc))]
+                     (let [id (:id node-desc-for-layout)
+                           node-desc-for-default-layout (id->node-desc-for-default-layout id)]
+                       (assert (map? node-desc-for-default-layout))
+                       (when (not= node-desc-for-default-layout node-desc-for-layout)
+                         node-desc-for-layout)))))))))
```

Comment:
The fix: We were only looking at property overrides to determine if we need to include a layout-specific `NodeDesc` for the node in the build output. However, template nodes apply their properties to the imported children before removing themselves from the built hierarchy, and we weren't seeing those property changes. In the new code, we apply the template flattening before this and compare the output of the layout-specific `NodeDesc` to that instead,